### ISSUE TITLE
Tox fails

### ISSUE THREAD

**Issue by lxsli** [gevent/gevent#795]
When I run tox, tests fail. I'm sure this is something I've done wrong but I've followed the README instructions to the best of my ability. On a fairly fresh linode instance, I have installed:

libffi-dev_3.1-2+b2_amd64.deb
python-cffi-dbg_0.8.6-1_amd64.deb
python-dbg_2.7.9-1_amd64.deb
python2.7-dbg_2.7.9-2_amd64.deb
libpython-dbg_2.7.9-1_amd64.deb
libpython2.7-dbg_2.7.9-2_amd64.deb
python-ply_3.4-5_all.deb
python-pycparser_2.10+dfsg-3_all.deb
python-cffi_0.8.6-1_amd64.deb
libexpat1_2.1.0-6+deb8u1_amd64.deb
libpython2.7_2.7.9-2_amd64.deb
libexpat1-dev_2.1.0-6+deb8u1_amd64.deb
libpython2.7-dev_2.7.9-2_amd64.deb
libpython-dev_2.7.9-1_amd64.deb
python2.7-dev_2.7.9-2_amd64.deb
python-dev_2.7.9-1_amd64.deb

and run `pip install -r dev-requirements.txt` (snipped a bunch of already satisfied reqs):

Collecting futures (from -r dev-requirements.txt (line 9))
  Using cached futures-3.0.5-py2-none-any.whl
Collecting restview (from -r dev-requirements.txt (line 13))
  Using cached restview-2.6.1-py2.py3-none-any.whl
Collecting readme-renderer (from restview->-r dev-requirements.txt (line 13))
  Using cached readme_renderer-0.7.0-py2.py3-none-any.whl
Collecting bleach (from readme-renderer->restview->-r dev-requirements.txt (line 13))
  Using cached bleach-1.4.2-py2.py3-none-any.whl
Collecting html5lib>=0.999 (from bleach->readme-renderer->restview->-r dev-requirements.txt (line 13))
Installing collected packages: futures, html5lib, bleach, readme-renderer, restview, gevent
  Running setup.py develop for gevent
Successfully installed bleach-1.4.2 futures-3.0.5 gevent html5lib-0.9999999 readme-renderer-0.7.0 restview-2.6.1

before running `tox -epy27`. Tox log [linked](http://pastebin.com/GSPQTjN2). I have latest pip, tox + virtualenv:

Successfully installed pip-8.1.1 setuptools-20.9.0 wheel-0.29.0
Successfully installed pluggy-0.3.1 py-1.4.31 tox-2.3.1 virtualenv-15.0.1

**Comment by jamadden** [comment]
A lot of the errors appear to be due to missing the standard Python `test` package. Depending on how you installed Python and your distribution, you may need to install a separate package to be able to run the standard Python tests (gevent runs the standard Python tests to be sure that monkey-patching works correctly). For example, on Debian-based systems, you need to install `libpython2.7-testsuite`. (I *just* ran into this myself and added a note to the README yesterday. Why do distributers feel the need to mess with standard Python? /)

**Comment by lxsli** [comment]
Thanks for the tip but I still get test failures. I also get failures on my workstation which has been running OpenStack unit tests for months.

[new tox log](http://pastebin.com/V9cBsPPK)

**Comment by lxsli** [comment]
Successfully installed test-2.3.4.5
Preparing to unpack .../libpython2.7-testsuite_2.7.9-2_all.deb ...

**Comment by jamadden** [comment]
Spot checking the remaining errors shows that they look like a mismatch between the version of the tests shipped (2.7.11) and the version of the standard library installed (purportedly 2.7.9). That and/or the distro has been messing with the modules (that's not uncommon for, for example, the ssl module---did your distro implement all of PEP 466; other vendors implemented it in a patch to 2.7.8 without changing the version number?)

**Comment by jamadden** [comment]
You know what, I ran into the same failures on Raspbian. Our Travis CI build runs [a 2.7.8 Python using *almost all* of the 2.7.11 tests](https://travis-ci.org/gevent/gevent/jobs/123047154#L283) (the only one overridden [is test_ssl.py](https://travis-ci.org/gevent/gevent/jobs/123047154#L284); tests in 2.7.8/ override those from 2.7/ in this case) without any of these problems. My conclusion was that the Debian packages are broken.

**Comment by lxsli** [comment]
$ cat /etc/os-release 
 PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
 NAME="Debian GNU/Linux"
 VERSION_ID="8"
 VERSION="8 (jessie)"
 ID=debian

Does that help at all? Aha I see your next comment. Maybe that's the issue. Let me try again on my Ubuntu 14.04.3 LTS workstation.

**Comment by lxsli** [comment]
Setting up libpython2.7-testsuite (2.7.6-8ubuntu0.2) ...

http://pastebin.com/VnmsWuzJ

Thanks for being so responsive on this! I have to go home now, I'll check back in on Monday.

**Comment by jamadden** [comment]
Ok, that took us all the way back to 2.7.6, which is not specifically tested. Therefore, I would expect test_ssl.py to fail...and it does.

- test_ssl.py fails because Python doesn't support the latest PEP 433 and there's no specific 2.7.6 version of that test installed.
- test___example_servers.py fails because you're apparently behind a proxy. A number of gevent's functional tests depend on network access.
- test__makefile_ref.py may be having trouble getting a list of the open file descriptors. Try uninstalling psutil (and make sure that lsof is installed and operational given an arbitrary pid).
- test__server.py can't resolve `''` to an address. Perhaps IPV6 is installed but not properly operational?
- test__socket_ssl.py is also making a network connection, which appears to be denied.

**Comment by lxsli** [comment]
Thanks for the help
